src/clients/SpokePoolClient.ts

```diff
@@ -298,24 +305,33 @@ export class SpokePoolClient extends BaseAbstractClient {
    * @returns A new deposit instance with the speed up signature appended to the deposit.
    */
   public appendMaxSpeedUpSignatureToDeposit(deposit: DepositWithBlock): DepositWithBlock {
-    const maxSpeedUp = this.speedUps[deposit.depositor]?.[deposit.depositId]?.reduce((prev, current) =>
-      prev.newRelayerFeePct.gt(current.newRelayerFeePct) ? prev : current
-    );
+    const { depositId, depositor } = deposit;
 
-    // We assume that the depositor authorises SpeedUps in isolation of each other, which keeps the relayer
-    // logic simple: find the SpeedUp with the highest relayerFeePct, and use all of its fields
-    if (!maxSpeedUp || maxSpeedUp.newRelayerFeePct.lte(deposit.relayerFeePct)) {
-      return deposit;
-    }
+    if (isV2Deposit(deposit)) {
+      const v2SpeedUps = (this.speedUps[depositor]?.[depositId]?.filter(() => isV2SpeedUp) ?? []) as v2SpeedUp[];
```

Comment:
```suggestion
      const v2SpeedUps = (this.speedUps[depositor]?.[depositId]?.filter(isV2SpeedUp) ?? []) as v2SpeedUp[];
```
Do you think this will work as it stands? We're not passing anything from the filter & we're not invoking the function.